app/gateway/proxy.py

```diff
@@ -80,35 +81,39 @@ def pass_through(path):
     # TODO: This switch should be resource dependent.
     if path.startswith('storage'):
         url = app.config['RENKU_ENDPOINT'] + "/api/" + path
+        auth_headers = headers
     else:
         url = app.config['GITLAB_URL'] + "/api/" + path
 
```

Comment:
Why is this part moved in the 'else' clause? Is there no need for authentication on the storage part?